src/ConnectedMode/Binding/CompositeBindingConfigProvider.cs

```diff
@@ -20,31 +20,39 @@
 
 using System;
 using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Diagnostics;
 using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
 using SonarLint.VisualStudio.Core;
 using SonarLint.VisualStudio.Core.Binding;
+using SonarLint.VisualStudio.Integration;
+using SonarQube.Client;
 using SonarQube.Client.Models;
 
 namespace SonarLint.VisualStudio.ConnectedMode.Binding
 {
-    public class CompositeBindingConfigProvider : IBindingConfigProvider
+    /// <summary>
+    /// Composite that coordinates turning server Quality Profile information into a set of config files
+    /// for a particular language
+    /// </summary>
+    [Export(typeof(IBindingConfigProvider))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    internal class CompositeBindingConfigProvider : IBindingConfigProvider
     {
         private readonly HashSet<IBindingConfigProvider> providers;
 
-        public CompositeBindingConfigProvider(params IBindingConfigProvider[] providers)
-        {
-            // params args can't be null - will be an empty array
-            if (providers.Length == 0)
-            {
-                throw new ArgumentNullException(nameof(providers));
-            }
-            if (providers.Any(p => p == null))
-            {
-                throw new ArgumentNullException(nameof(providers));
-            }
+        [ImportingConstructor]
+        public CompositeBindingConfigProvider(ISonarQubeService sonarQubeService, ILogger logger)
+            : this(
+                  new CSharpVBBindingConfigProvider(sonarQubeService, logger),
+                  new NonRoslynBindingConfigProvider(sonarQubeService, logger))
```

Comment:
I could have made the `CSharpVBBindingConfigProvider` and `NonRoslynBindingConfigProvider` providers MEF-creatable and done an `ImportMany` here, but it didn't seem worth it.

It would have meant defining a new interface and writing more MEF tests for no real gain.
The list of config providers we support is fixed, the `CompositeBindingConfigProvider` is the only thing that needs to use them, and they are easy to create. It was simpler just to make the composite provider responsible for creating them.

If we do need to add other types of config provider in the future it would be easy enough to change later.